ncm-dpmlfc/src/main/perl/dpmlfc/RuleBasedEditor.pm

```diff
@@ -0,0 +1,817 @@
+    $self->debug(2,"$function_name: configuration value '$config_value' will be added to the pattern");
+    $config_value =~ s/\\/\\\\/g;
+    $config_value =~ s/([\-\+\?\.\*\[\]()\^\$])/\\$1/g;
+    $config_value =~ s/\s+/\\s+/g;
+  } else {
+    $config_value = "";
+  }
+
+  # config_param is generally a keyword and in this case it contains no whitespace.
+  # A special case is when config_param (the rule keyword) is used to match a line
+  # without specifying a rule: in this case it may contains whitespaces. Remove strict
+  # matching of them (match any type/number of whitespaces at the same position). 
+  # Look at %trust_config_rules in ncm-dpmlfc Perl module for an example.
+  $config_param =~ s/\s+/\\s+/g;
+
+  my $config_param_pattern;
+  if ( $line_fmt == LINE_FORMAT_PARAM ) {
+    $config_param_pattern = "#?\\s*$config_param=".$config_value;
+  } elsif ( $line_fmt == LINE_FORMAT_ENVVAR ) {
+    $config_param_pattern = "#?\\s*export $config_param=".$config_value;
+  } elsif ( $line_fmt == LINE_FORMAT_XRDCFG_SETENV ) {
+    $config_param_pattern = "#?\\s*setenv\\s+$config_param\\s*=\\s*".$config_value;
+  } elsif ( $line_fmt == LINE_FORMAT_XRDCFG_SET ) {
+    $config_param_pattern = "#?\\s*set\\s+$config_param\\s*=\\s*".$config_value;
+  } elsif ( $line_fmt == LINE_FORMAT_XRDCFG ) {
+    $config_param_pattern = "#?\\s*$config_param";
+    # Avoid adding a whitespace requirement if there is no config_value
+    if ( $config_value ne "" ) {
+      $config_param_pattern .= "\\s+" . $config_value;
+    }
+  } else {
+    $self->error("$function_name: invalid line format ($line_fmt). Internal inconsistency.");
+    return undef;
+  }
+
+  return $config_param_pattern
+}
+
+
+=pod
+
+=item _removeConfigLine
+
+This function comments out a configuration line matching the configuration parameter.
+Match operation takes into account the line format.
+
+Arguments :
+    fh : a FileEditor object
+    config_param: parameter to update
+    line_fmt : line format (see LINE_FORMAT_xxx constants)
+
+=cut
+
+sub _removeConfigLine {
+  my $function_name = "_removeConfigLine";
+  my ($self, $fh, $config_param, $line_fmt) = @_;
+
+  unless ( $fh ) {
+    $self->error("$function_name: 'fh' argument missing (internal error)");
+    return 1;
+  }
+  unless ( $config_param ) {
+    $self->error("$function_name: 'config_param' argument missing (internal error)");
+    return 1;
+  }
+  unless ( defined($line_fmt) ) {
+    $self->error("$function_name: 'line_fmt' argument missing (internal error)");
+    return 1;
+  }
+
+  # Build a pattern to look for.
+  my $config_param_pattern = $self->_buildLinePattern($config_param,$line_fmt);
+
+  $self->debug(1,"$function_name: commenting out lines matching pattern >>>".$config_param_pattern."<<<");
+  # All matching lines must be commented out, except if they are already commented out.
+  # The code used is a customized version of FileEditor::replace() that lacks support for backreferences
```

Comment:
open an issue with an example, i still don't fully see the case.
but shouldn't be too hard to add and if it's useful, i see no point why not.